### PR TITLE
#538 customer requisition detail view

### LINKED ISSUE #538: Customer requisition detail view
Epic: #100 

Add the customer requisition detail view

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #538 

- Basic copy paste of the supplier requisition detail view. I removed some columns for now. The customer requisition columns are a bit confusing so thought to make an issue specifically for that

### CODE REVIEW

**Comment by joshxg** [comment]
jeez these prs really get big

File: packages/requisitions/src/CustomerRequisition/DetailView/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            height={64}
+          >
+            <StatusCrumbs
+              statuses={getSupplierRequisitionStatuses()}
+              statusLog={createStatusLog(draft.status)}
+              statusFormatter={getSupplierRequisitionTranslator()}
+            />
+
+            <Box flex={1} display="flex" justifyContent="flex-end" gap={2}>
+              <ButtonWithIcon
```
Comment (mark-prins) [inline packages/requisitions/src/CustomerRequisition/DetailView/Footer.tsx:93]: now that this has been replicated a few times.. maybe we should add a button type which has secondary colouring and small font size and `shrinkThreshold` all defaulting to useful values?

File: packages/requisitions/src/types.ts [hunk 2]
```diff
@@ -37,9 +37,25 @@ export interface SupplierRequisition extends Requisition {
   deleteLine?: (line: SupplierRequisitionLine) => void;
 }
 
+export interface CustomerRequisition extends Requisition {
+  isDeleted: boolean;
+  otherParty: Name;
+  lines: SupplierRequisitionLine[];
```
Comment (mark-prins) [inline packages/requisitions/src/types.ts:43]: should be `CustomerRequisitionLine` for `lines` and the mutations below

File: packages/requisitions/src/utils.ts [hunk 3]
```diff
@@ -29,9 +33,35 @@ export const placeholderSupplierRequisition: SupplierRequisition = {
...
+  otherParty: {
+    id: '',
+    name: '',
+    code: '',
+    isCustomer: false,
+    isSupplier: true,
+  },
+  otherPartyId: '',
+  otherPartyName: '',
+  status: SupplierRequisitionNodeStatus.Draft,
```
Comment (mark-prins) [inline packages/requisitions/src/utils.ts:51]: had a note in #540 about what to do with the status - currently `SupplierRequisitionNodeStatus` for both types of req

**Review by mark-prins — APPROVED** [review]
👍  awesome!

File: packages/requisitions/src/CustomerRequisition/DetailView/Footer.tsx (hunk 1 above)
Comment (joshxg) [reply packages/requisitions/src/CustomerRequisition/DetailView/Footer.tsx:93]: Ah good call. I supposed it's the same as the `dialog button` style? 🤔

File: packages/requisitions/src/types.ts (hunk 2 above)
Comment (joshxg) [reply packages/requisitions/src/types.ts:43]: thanks!

**Pull request merged**